src/Microsoft.AspNet.Mvc.Core/DefaultControllerFactory.cs

```diff
@@ -38,8 +38,19 @@ public DefaultControllerFactory(IControllerActivator controllerActivator)
             _getPropertiesToActivate = GetPropertiesToActivate;
         }
 
+        /// <summary>
+        /// The <see cref="IControllerActivator"/> used to create a controller.
+        /// </summary>
+        protected IControllerActivator ControllerActivator
```

Comment:
I am wondering ControllerActivator is a transient service, while ControllerFactory is a singleton are we not increasing the lifetime  of the activator?